### PR TITLE
[IMP] pos: change default parameters for Tips and Discount products

### PR DESCRIPTION
**Pull request by loukwets**
[IMP] point_of_sale: change default parameters Tips Discount products

- Unable to do global discounts because the `Discount` product isn't
`Available in PoS`. The available in pos paramater of the `Discount`
product in the data has then been set to true.

- The attribute `tip_product_id` gets as default the product of default_code TIPS.

- The condition of the display of tips option in pos payement has been
changed to the boolean corresponding to the tick box of the settings and
the `tip_product_id`.

- Before, if a tip product was set and the tips
option in the setting unselected the tips was still availible in the pos
payement since it was based on the tip product id and not on the tick
box. However, the `tip_product_id` still remains in the condition since
it can be possible to check the option and not select a tip product.

Task-3283007
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by caburj** [comment]
When putting the name of the addon in the commit title, don't use "pos". Use "point_of_sale".

File: addons/point_of_sale/models/pos_config.py [hunk 1]
```diff
@@ -46,6 +46,9 @@ def _get_group_pos_manager(self):
     def _get_group_pos_user(self):
         return self.env.ref('point_of_sale.group_pos_user')
 
+    def _default_tip_product_id(self):
+        return self.env['product.product'].search([('default_code', '=', 'TIPS')])
```
Comment (davidmonnom) [inline addons/point_of_sale/models/pos_config.py:50]: I don't know if this is very relevant but isn't better to use the `xmlId` instead of the `default_code` ? We can see with @caburj. 

The default code is `product_product_tip`

Comment (davidmonnom) [reply addons/point_of_sale/models/pos_config.py:50]: Or maybe not, I don't really know what to do in that case, I'm interested to have the answer.

Comment (caburj) [reply addons/point_of_sale/models/pos_config.py:50]: @loukwets @davidmonnom  xml ID points to a record. But this record is not immune from deletion. So, if you try to get the value of that xml ID using `env.ref`, it's possible that you don't get a record. But indeed, MODA has a good point. Since we have "product_product_tip" as part of data set when installing pos, we can use that by default. But if not found, we look for other TIPS product as default.

Comment (caburj) [reply addons/point_of_sale/models/pos_config.py:50]: One more thing. Only take the first item in the search result otherwise it might crash because tip_product_id is many2one.

File: addons/point_of_sale/models/res_config_settings.py [hunk 2]
```diff
@@ -247,7 +247,7 @@ def _compute_pos_fiscal_positions(self):
     def _compute_pos_tip_product_id(self):
         for res_config in self:
             if res_config.pos_iface_tipproduct:
-                res_config.pos_tip_product_id = res_config.pos_config_id.tip_product_id
+                res_config.pos_tip_product_id = self.env['product.product'].search([('default_code', '=', 'TIPS')])
```
[the remark this answers is not in the document]
Comment (loukwets) [reply addons/point_of_sale/models/res_config_settings.py:250]: If I rember well the default tips product in pos.config model is only set when the model is intalled, if the tip product is created afterwards it won't be set in `res_config.pos_config_id.tip_product_id`

[the remark this answers is not in the document]
Comment (caburj) [reply addons/point_of_sale/models/res_config_settings.py:250]: we should keep the `res.config.settings` this way. If you want to conditionally put a value in the tip_product_id, do it in `write` method of the `pos.config` model.

[the remark this answers is not in the document]
Comment (caburj) [reply addons/point_of_sale/models/res_config_settings.py:250]: > we should keep the res.config.settings this way.

Why? Because "res.config.settings" is the "temporary" place where we modify the "pos.config". As much as possible, we still put the business logic in "pos.config". So, I'd suggest to put the logic of setting default tips in the "pos.config" model.

File: addons/point_of_sale/data/point_of_sale_data.xml [hunk 3]
```diff
@@ -30,7 +30,7 @@
 
         <record id="product_product_consumable" model="product.product">
             <field name="name">Discount</field>
-            <field name="available_in_pos">False</field>
+            <field name="available_in_pos">True</field>
```
Comment (caburj) [inline addons/point_of_sale/data/point_of_sale_data.xml:33]: While fixing the commit message, can you also include why this change is need?

**Review by caburj — CHANGES_REQUESTED** [review]
Can you properly write the commit message by listing the objectives? I think we are good to go with this PR.

File: addons/point_of_sale/static/src/xml/Screens/PaymentScreen/PaymentScreen.xml [hunk 4]
```diff
@@ -87,7 +87,7 @@
                                     t-on-click="toggleIsToInvoice">
                                     <i class="fa fa-file-text-o" /> Invoice
                                 </div>
-                                <div t-if="env.pos.config.tip_product_id" class="button js_tip"
+                                <div t-if="env.pos.config.iface_tipproduct and env.pos.config.tip_product_id" class="button js_tip"
```
Comment (caburj) [inline addons/point_of_sale/static/src/xml/Screens/PaymentScreen/PaymentScreen.xml:90]: You need to rebase. During rebase, you'll have to replace the t-if with `pos.globalState.config.iface_tipproduct and pos.globalState.config.tip_product_id`.

File: addons/point_of_sale/static/src/xml/Screens/PaymentScreen/PaymentScreen.xml [hunk 5]
```diff
@@ -87,7 +87,7 @@
                                     t-on-click="toggleIsToInvoice">
                                     <i class="fa fa-file-text-o" /> Invoice
                                 </div>
-                                <div t-if="pos.globalState.config.tip_product_id" class="button js_tip"
+                                <div t-if="env.pos.config.iface_tipproduct and env.pos.config.tip_product_id" class="button js_tip"
```
Comment (caburj) [inline addons/point_of_sale/static/src/xml/Screens/PaymentScreen/PaymentScreen.xml:90]: replace the expression inside t-if with `pos.globalState.config.iface_tipproduct and pos.globalState.config.tip_product_id`. `env.pos` doesn't work anymore in master because of this: https://github.com/odoo/odoo/pull/120992
